src/platform/site-wide/side-nav/components/SideNav.js

```diff
@@ -46,6 +46,28 @@ class SideNav extends Component {
     });
   };
 
+  /*
+   Detects expanded children to modify the ending line style
+   with a short delay.
+   This is looks a bit tricky but ending will have to render always first
+   for level 1 items and then because of the recursive way of rendering children
+   after this we can detect that children has been expanded and apply
+   the style by finding it by its parentID.
+  */
+  shouldModifyEndingLine = (item, depth) => {
+    const expanded = get(item, 'expanded');
+    const hasChildren = get(item, 'hasChildren');
+    const shouldShowLineOpen = !!(depth === 2 && expanded && hasChildren);
+    if (shouldShowLineOpen) {
+      setTimeout(() => {
```

Comment:
I think we may want to explore a different solution here so that we don't have to deal with adding timeouts = navItems.length. I think within the `<NavItem />` component we can optionally add this className if `depth === 2 && expanded && hasChildren` 🙂

Reply:
The problem is that I am trying to produce exactly what the wireframes require  (no separator line when items expanded ): 

<img width="293" alt="Screen Shot 2020-04-28 at 3 57 22 PM" src="https://user-images.githubusercontent.com/100468/80537101-167bb980-8969-11ea-9902-23ea32ad7714.png">

But yes initially when I thought about this logic, should be pretty straight forward something like:

```
const showEndingLine = (item, depth, sortedNavItems, index) => {
  const expanded = get(item, 'expanded');
  const hasChildren = get(item, 'hasChildren');
  const shouldShowLineOpen = !!(depth === 2 && expanded && hasChildren);
  const isFirstLevel = depth === 1;
  const isLastNavItem = index === sortedNavItems.length - 1;
  if (shouldShowLineOpen) {
    return <div className="line-open" />;
  } else if (isFirstLevel && !isLastNavItem) {
    return <div className="line" />;
  }
  return null;
};
```

which shows something like this ending line + separator line. This is why I mentioned in the fn that modifies the className that it is tricky to know when to render these lines according to the states/level.

The separator line logic applies always because `depth === 2 && expanded && hasChildren` in the first recursive is false will not take over the separator logic.
I might be overthinking so, I prefer to have more feedback from the designers first and I will point that out so that we don't have to add those delays to modify the class names.

<img width="786" alt="Screen Shot 2020-04-28 at 3 24 18 PM" src="https://user-images.githubusercontent.com/100468/80537400-9c980000-8969-11ea-80ef-e8348aca4b7c.png">